packages/db-ivm/src/operators/topKWithFractionalIndex.ts

```diff
@@ -261,30 +258,28 @@ export class TopKWithFractionalIndexOperator<K, T> extends UnaryOperator<
     super(id, inputA, output)
     const limit = options.limit ?? Infinity
     const offset = options.offset ?? 0
-    const compareTaggedValues = (
-      a: TaggedValue<K, T>,
-      b: TaggedValue<K, T>
-    ) => {
+    const compareKeyedValues = ([aKey, aVal]: [K, T], [bKey, bVal]: [K, T]) => {
       // First compare on the value
-      const valueComparison = comparator(getVal(a), getVal(b))
+      const valueComparison = comparator(aVal, bVal)
       if (valueComparison !== 0) {
         return valueComparison
       }
-      // If the values are equal, compare on the tag (object identity)
-      const tieBreakerA = getTag(a)
-      const tieBreakerB = getTag(b)
-      return tieBreakerA - tieBreakerB
+      // If the values are equal, use the row key as tie-breaker
+      // This provides stable, deterministic ordering since keys are string | number
+      if (aKey === bKey) return 0
```

Comment:
This tie-breaking doesn't work if `aKey` and `bKey` have different types, e.g. `'a' < 5` and `5 < 'a'` and `5 == 'a'` all evaluate to false.
We need to keep into account their type:
```ts
if (typeof aKey == typeof bKey) {
  // current comparison
} else if (typeof aKey === 'number') {
  return -1
} else {
  // it's a string
  return 1
}
```